### PR TITLE
conflict_resolver: give up if too many attempts

### PR DESCRIPTION
**Pull request by jzila**
Port from keybase/kbfs#1995

Give up if CR has failed 5 times in a row.

Issue: KBFS-3681

@jakob223

### CODE REVIEW

File: go/kbfs/libkbfs/conflict_resolver.go [hunk 1]
```diff
@@ -3325,10 +3330,16 @@ func (cr *ConflictResolver) doResolve(ctx context.Context, ci conflictInput) {
...
+	var failCount int
+	failCount, err = cr.recordStartResolve(ci)
 	if err != nil {
 		cr.log.CWarningf(ctx,
 			"Could not record conflict resolution attempt: %v", err)
+	} else if failCount >= maxConflictResolutionAttempts {
+		cr.log.CWarningf(ctx,
+			"Maximum number of conflict resolution attempts reached for TLF "+
+				"%s: auto failing.", cr.fbo.id().String())
+
```
Comment (strib) [inline go/kbfs/libkbfs/conflict_resolver.go]: Extra blank line.

File: go/kbfs/libkbfs/conflict_resolver.go [hunk 2]
```diff
@@ -3325,10 +3330,16 @@ func (cr *ConflictResolver) doResolve(ctx context.Context, ci conflictInput) {
...
 	defer func() { cr.config.MaybeFinishTrace(ctx, err) }()
 
-	err = cr.recordStartResolve(ci)
+	var failCount int
+	failCount, err = cr.recordStartResolve(ci)
 	if err != nil {
 		cr.log.CWarningf(ctx,
 			"Could not record conflict resolution attempt: %v", err)
+	} else if failCount >= maxConflictResolutionAttempts {
+		cr.log.CWarningf(ctx,
```
Comment (strib) [inline go/kbfs/libkbfs/conflict_resolver.go]: Did you intend to do something other than print a warning message here?  Like maybe returning?

Comment (jakob223) [reply go/kbfs/libkbfs/conflict_resolver.go]: I did intend to return here.

**Comment by jakob223** [comment]
closing in favor of https://github.com/keybase/client/pull/15409 since that's easier than rebasing this

**Pull request closed without merging**